modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/helper/SPAHelper.java

```diff
@@ -175,139 +195,67 @@ public boolean isDebugEnabled() {
-
-	@Activate
-	protected void activate(
-			BundleContext bundleContext, Map<String, Object> properties)
-		throws InvalidSyntaxException {
-
-		_spaConfiguration = ConfigurableUtil.createConfigurable(
-			SPAConfiguration.class, properties);
-
-		_cacheExpirationTime = _getCacheExpirationTime(_spaConfiguration);
-		_spaExcludedPathsJSONArray = _getExcludedPathsJSONArray(
-			_spaConfiguration);
-
-		_spaExcludedTargetPortletsJSONArray =
-			_getExcludedTargetPortletsJSONArray();
-
-		Collections.addAll(
-			_navigationExceptionSelectors,
-			_spaConfiguration.navigationExceptionSelectors());
-
-		_navigationExceptionSelectorsString = ListUtil.toString(
-			_navigationExceptionSelectors, (String)null, StringPool.BLANK);
-
-		Filter filter = bundleContext.createFilter(
-			"(&(objectClass=java.lang.Object)(" +
-				_SPA_NAVIGATION_EXCEPTION_SELECTOR_KEY + "=*))");
-
-		_navigationExceptionSelectorTracker = new ServiceTracker<>(
-			bundleContext, filter,
-			new NavigationExceptionSelectorTrackerCustomizer(bundleContext));
-
-		_navigationExceptionSelectorTracker.open();
-
-		_serviceRegistration = bundleContext.registerService(
-			Consumer.class,
-			companyId -> {
-				if (companyId == null) {
-					_portletsBlacklistJSONArrays.clear();
-				}
-				else {
-					_portletsBlacklistJSONArrays.remove(companyId);
-				}
-			},
-			MapUtil.singletonDictionary(
-				"portlets.map.clear.consumer", Boolean.TRUE));
-	}
-
-	@Deactivate
-	protected void deactivate() {
-		_serviceRegistration.unregister();
-		_navigationExceptionSelectorTracker.close();
+	public boolean isEnabled() {
+		return _spaConfiguration.enabled();
 	}
 
-	@Modified
-	protected void modified(Map<String, Object> properties) {
-		_navigationExceptionSelectors.removeAll(
-			Arrays.asList(_spaConfiguration.navigationExceptionSelectors()));
-
-		_spaConfiguration = ConfigurableUtil.createConfigurable(
-			SPAConfiguration.class, properties);
-
-		_cacheExpirationTime = _getCacheExpirationTime(_spaConfiguration);
-		_spaExcludedPathsJSONArray = _getExcludedPathsJSONArray(
-			_spaConfiguration);
-
-		_spaExcludedTargetPortletsJSONArray =
-			_getExcludedTargetPortletsJSONArray();
-
-		Collections.addAll(
-			_navigationExceptionSelectors,
-			_spaConfiguration.navigationExceptionSelectors());
-
-		_navigationExceptionSelectorsString = ListUtil.toString(
-			_navigationExceptionSelectors, (String)null, StringPool.BLANK);
+	public boolean isPreloadCSS() {
+		return _spaConfiguration.preloadCSS();
 	}
 
-	private long _getCacheExpirationTime(SPAConfiguration spaConfiguration) {
-		long cacheExpirationTime = spaConfiguration.cacheExpirationTime();
-
-		if (cacheExpirationTime > 0) {
-			cacheExpirationTime *= Time.MINUTE;
-		}
-
-		return cacheExpirationTime;
-	}
+	private static final String _REDIRECT_PARAM_NAME;
 
-	private JSONArray _getExcludedPathsJSONArray(
-		SPAConfiguration spaConfiguration) {
+	private static final SPAConfiguration _SPA_CONFIGURATION =
+		new SPAConfiguration() {
```

Comment:
Is [Iván's comment](https://github.com/liferay-frontend/liferay-portal/pull/4948/files#r2126742498) applicable here too?